app/controllers/authenticate.js

```diff
@@ -11,12 +11,28 @@ export default GoodcityController.extend(preloadDataMixin, {
   authService: Ember.inject.service(),
   attemptedTransition: null,
   pin: "",
+  otpResendTime: config.APP.OTP_RESEND_TIME,
+  pinAlreadySent: false,
   isMobileApp: config.cordova.enabled,
 
   mobile: Ember.computed("mobilePhone", function() {
     return config.APP.HK_COUNTRY_CODE + this.get("mobilePhone");
   }),
 
+  timerFunction() {
+    let waitTime = this.get("otpResendTime");
+    if (waitTime == 0) {
+      this.set("pinAlreadySent", false);
+      this.set("otpResendTime", config.APP.OTP_RESEND_TIME);
+      return false;
+    }
+    this.set("timer", waitTime);
+    this.set("otpResendTime", waitTime - 1);
+    return setTimeout(() => {
+      this.timerFunction();
+    }, 1000);
+  },
```

Comment:
This looks alright, however I'm not sure why you need both `timer` and `otpResendTime` (can probably make do with a single one)

Aslo you should probably cancel the timeout when you navigate away from the page